app/code/community/Reverb/ReverbSync/Model/Observer.php

```diff
@@ -7,11 +7,24 @@ class Reverb_ReverbSync_Model_Observer
     //function to create the product in reverb
     public function productSave($observer)
     {
-        $productSyncHelper = Mage::helper('ReverbSync/sync_product');
-        $product_id = $observer->getProduct()->getId();
-        try
-        {
-            $listingWrapper = $productSyncHelper->executeIndividualProductDataSync($product_id);
+      try
+      {
+          $product = $observer->getProduct();
+          $product_id = $product->getId();
+
+          $syncToReverb = $product->getData('rev_sync');
+          if (is_null($syncToReverb)) {
+              // TODO: This will potentially generate one SQL query per product saved.  Revamp this to queue the check and process in async batches.
+              $syncToReverb = Mage::getResourceModel('catalog/product')->getAttributeRawValue($product_id, 'rev_sync');
```

Comment:
This is true, often-times a product will be loaded with only a subset of its attributes; as such we can not always rely on the product having been fully loaded with the rev_sync attribute at this point.

Steven's call to getAttributeRawValue() selects the rev_sync value for the product with an adapter query, avoiding a number of ORM related event-observers which likely are unnecessary in terms of loading the rev_sync attribute. 

By returning false (as opposed to throwing a Reverb_ReverbSync_Model_Exception_Product_Excluded exception) is that no messaging will be shown to the admin that the product was not synced due to this setting. This is a business level decision on whether or not to assume the admin user knows of that setting and knows that the product isn't being synced.